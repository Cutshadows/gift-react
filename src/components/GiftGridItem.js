import React from 'react';
const GiftGridItem = ({title, url}) => {
   // const {url, id, title}=img;
    return ( 
        <div className="card animate__animated animate__pulse">
            <img src={url} alt={title}/>
            <p>{title.substring(0, 12)}</p>
        </div>
     );
}
 
export default GiftGridItem;