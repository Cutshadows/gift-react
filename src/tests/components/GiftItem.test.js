import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';

// import GiftExpertApp from '../components/GiftExpertApp';

import GiftGridItem from '../../components/GiftGridItem';

describe('Pruebas en <GiftExpertApp></GiftExpertApp>', () => {
   // let wrapper=shallow(<GiftGridItem />);
   /*  beforeEach(()=>{
        wrapper=shallow(<GiftGridItem/>);
    }); */
    const title='Un title';
    const url='https://localhost/algo.jpg';
    let wrapper=shallow(
        <GiftGridItem 
        title={title} 
        url={url} />
        );


    test('debe mostart <GiftGridItem></GiftGridItem> Correctamente ', () => {
            expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un parrafo con el title', () => {
        const p=wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe de tner la img igual al url y alt de los props', () => {
        const img=wrapper.find('img');

        expect( img.prop('src') ).toBe(url);//props().src html 

        expect( img.prop('alt') ).toBe(title);//props().src html 

    })
    
    test('debe de tener animate__pulse', () => {
         const div=wrapper.find('div');
       /* expect(div.hasClass('animate__pulse')).toBeTruthy() */
        //    console.log(div.props())
        const className=div.prop('className');
       expect(className.includes('animate__pulse')).toBe(true);
    })
    
    
    
})
