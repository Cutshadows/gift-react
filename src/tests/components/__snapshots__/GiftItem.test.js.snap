// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp></GiftExpertApp> debe mostart <GiftGridItem></GiftGridItem> Correctamente  1`] = `
<div
  className="card animate__animated animate__pulse"
>
  <img
    alt="Un title"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un title
  </p>
</div>
`;
